sycl/doc/developer/KHRExtensions.md

```diff
@@ -0,0 +1,46 @@
+# Considerations for working on KHR extensions
+
+SYCL specification evolves through embedding extensions developed by various
+vendors, including Khronos Group itself (`khr` extensions).
+
+In order for a KHR extension to be accepted, there must be CTS tests for it and
+at least one implementation which passes them.
+
+Considering that KHR extensions are being developed in public, we can start
+prototyping them as soon as corresponding PR for an extension is published at
+KhronosGroup/SYCL-Docs.
+
+However, we shouldn't be exposing those extensions to end users until the
+extension if finalised, ratified and published by Khronos - due to risk of an
+extension changing during that process and lack of the officially published
+version of it.
+
+So, we can have a PR but can't merge it. Keeping PRs opened for a long time is a
+bad practice, because they tend to get stale: there are merge conflicts,
+potential functional issues due to the codebase changes, etc.
+
+In order for us to avoid stale PRs, all functionality which is a public
+interface of an "in-progress" KHR extension, must be hidden under
+`__DPCPP_ENABLE_UNFINISHED_KHR_EXTENSIONS` macro. That way we can merge a PR to
+avoid constantly maintaining it in a good shape, start automatically testing it
+but at the same time avoid exposing incomplete and/or undocumented feature to
+end users just yet.
+
+"in-progress" KHR extension term used above is defined as:
+- KHR extension proposed to Khronos Group which has not been published in a
+  released revision of the SYCL specification (of any version).
+
+  Note: merge of an extension proposal PR into KhronosGroup/SYCL-Docs repo is
+  **not** considered to be a publishing event. The extension specification has
+  to be available on the Khronos SYCL Registry website:
+  https://registry.khronos.org/SYCL/
```

Comment:
I'm not sure about this part.  For changes to the SYCL specification, we usually consider a change to be "accepted" by the Khronos WG when the PR is merged to the SYCL-Docs repo.  Why not follow the same strategy for KHRs?

More precisely, I think we should wait for the KHR to be merged / cherry-picked to the "sycl-2020" branch of SYCL-Docs.  We don't do this until the KHR has been approved by the WG and also ratified by the Khronos board.

Reply:
> I'm not sure about this part. For changes to the SYCL specification, we usually consider a change to be "accepted" by the Khronos WG when the PR is merged to the SYCL-Docs repo. Why not follow the same strategy for KHRs?

I guess my point was also to wait until the documentation is publicly available to end users in some friendly form. But I agree that this is not a critical thing so I'm fine with different wording here.

> More precisely, I think we should wait for the KHR to be merged / cherry-picked to the "sycl-2020" branch of SYCL-Docs. We don't do this until the KHR has been approved by the WG and also ratified by the Khronos board.

Sure, I can apply that

Reply:
Reworded that bullet in 0588f7f3c8489296eeaa0d6771d4cfd18635e659